tests/integration/components/polaris-form-test.js

```diff
@@ -0,0 +1,175 @@
+      .exists('renders implicit submit button');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('method', method, 'renders method attribute when provided');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('name', name, 'renders name attribute when provided');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute(
+        'novalidate',
+        '',
+        'renders novalidate attribute when provided'
+      );
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('target', target, 'renders target attribute when provided');
+
+    this.set('implicitSubmit', false);
+
+    assert
+      .dom('[data-test-form-button]')
+      .doesNotExist(
+        'does not render submit button when `implicitSubmit` is false'
+      );
+  });
+
+  test('invokes `onSubmit` when form is submitted', async function(assert) {
+    assert.expect(1);
+
+    this.onSubmitSpy = () => {
+      assert.ok('invokes `onSubmit` action');
+    };
+
+    await render(hbs`{{polaris-form
+      onSubmit=(action onSubmitSpy)
+    }}`);
+
+    await triggerEvent('[data-test-form]', 'submit');
+  });
+
+  module('preventDefault', function() {
+    test('allows default handler to handle submit event when set to false', async function(assert) {
+      assert.expect(0);
+
+      this.onSubmitSpy = () => {
+        assert.notOk('does not invoke `onSubmit`');
+      };
+      this.preventDefaultSpy = () => {
+        assert.notOk('does not prevents default');
+      };
+
+      await render(hbs`
+        <form on-submit={{action preventDefaultSpy}}>
+          {{polaris-form
+            preventDefault=false
+            onSubmit=(action onSubmitSpy)
+          }}
+        </form>
+      `);
+
+      await triggerEvent('[data-test-form]', 'submit');
+    });
+
+    test('submits the form preventing default when not provided', async function(assert) {
+      assert.expect(0);
+
+      this.onSubmitSpy = () => {
+        assert.ok('invokes `onSubmit`');
+      };
+      this.preventDefaultSpy = () => {
+        assert.ok('prevents default `submit`');
+      };
+
+      await render(hbs`
+        <form on-submit={{action preventDefaultSpy}}>
+          {{polaris-form
+            onSubmit=(action onSubmitSpy)
+          }}
+        </form>
+      `);
+
+      await triggerEvent('[data-test-form]', 'submit');
+    });
+
+    test('submits the form preventing default when set to true', async function(assert) {
+      assert.expect(0);
+
+      this.onSubmitSpy = () => {
+        assert.ok('invokes `onSubmit`');
+      };
+      this.preventDefaultSpy = () => {
+        assert.ok('prevents default `submit`');
+      };
+
+      await render(hbs`
+        <form on-submit={{action preventDefaultSpy}}>
+          {{polaris-form
+            preventDefault=true
+            onSubmit=(action onSubmitSpy)
+          }}
+        </form>
+      `);
+
+      await triggerEvent('[data-test-form]', 'submit');
+    });
+  });
+});
```

Comment:
How do you feel about adding the same tests as the [React](https://github.com/Shopify/polaris/blob/master/src/components/Form/tests/Form.test.tsx) component has here?

Reply:
not sure which one do you refer to....I think all  of them are already covered here

Reply:
Oh, my bad then 😅